CI build failed for merged PR

https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/pipelines/14338

```
Traceback (most recent call last):
  File "/builds/espressomd/espresso/build/testsuite/scripts/tutorials/test_raspberry_electrophoresis.py", line 43, in test_trajectory_simulated
    self.assertGreater(dist[0], np.min(dist[1:]))
AssertionError: 2.169 not greater than 2.8839
```

--- a/testsuite/python/coulomb_cloud_wall.py
+++ b/testsuite/python/coulomb_cloud_wall.py
@@ -115,22 +115,6 @@ class CoulombCloudWall(ut.TestCase):
         self.S.integrator.run(0)
         self.compare("p3m_gpu", energy=False, prefactor=2.2)
 
-    @ut.skipIf(not espressomd.has_features(["SCAFACOS"])
-               or 'p3m' not in scafacos.available_methods(),
-               'Skipping test: missing feature SCAFACOS or p3m method')
-    def test_scafacos_p3m(self):
-        self.S.actors.add(
-            espressomd.electrostatics.Scafacos(
-                prefactor=0.5,
-                method_name="p3m",
-                method_params={
-                    "p3m_r_cut": 1.001,
-                    "p3m_grid": 64,
-                    "p3m_cao": 7,
-                    "p3m_alpha": 2.70746}))
-        self.S.integrator.run(0)
-        self.compare("scafacos_p3m", energy=True, prefactor=0.5)
-
     @ut.skipIf(not espressomd.has_features("SCAFACOS")
                or 'p2nfft' not in scafacos.available_methods(),
                'Skipping test: missing feature SCAFACOS or p2nfft method')

--- a/testsuite/python/dipolar_direct_summation.py
+++ b/testsuite/python/dipolar_direct_summation.py
@@ -198,7 +198,9 @@ class dds(ut.TestCase):
             force_tol=1E-4,
             torque_tol=1E-4)
 
-    @utx.skipIfMissingFeatures("SCAFACOS_DIPOLES")
+    @ut.skipIf(not espressomd.has_features("SCAFACOS_DIPOLES") or
+               "direct" not in espressomd.scafacos.available_methods(),
+               "Skipping test: missing SCAFACOS_DIPOLES or 'direct' method")
     def test_dds_scafacos(self):
         self.check_open_bc(
             self.fcs_data,

--- a/testsuite/python/save_checkpoint.py
+++ b/testsuite/python/save_checkpoint.py
@@ -244,7 +244,8 @@ if espressomd.has_features('DP3M') and 'DP3M' in modes:
         tune=False)
     system.actors.add(dp3m)
 
-if espressomd.has_features('SCAFACOS') and 'SCAFACOS' in modes:
+if espressomd.has_features('SCAFACOS') and 'SCAFACOS' in modes \
+        and 'p3m' in espressomd.scafacos.available_methods():
     system.actors.add(espressomd.electrostatics.Scafacos(
         prefactor=0.5,
         method_name="p3m",
@@ -254,7 +255,8 @@ if espressomd.has_features('SCAFACOS') and 'SCAFACOS' in modes:
             "p3m_cao": 7,
             "p3m_alpha": 2.084652}))
 
-if espressomd.has_features('SCAFACOS_DIPOLES') and 'SCAFACOS' in modes:
+if espressomd.has_features('SCAFACOS_DIPOLES') and 'SCAFACOS' in modes \
+        and 'p2nfft' in espressomd.scafacos.available_methods():
     system.actors.add(espressomd.magnetostatics.Scafacos(
         prefactor=1.2,
         method_name='p2nfft',

--- a/testsuite/python/scafacos_interface.py
+++ b/testsuite/python/scafacos_interface.py
@@ -50,6 +50,9 @@ class ScafacosInterface(ut.TestCase):
         for method in available_methods:
             self.assertIn(method, scafacos_methods)
 
+    @ut.skipIf(not espressomd.has_features('SCAFACOS') or
+               'p3m' not in espressomd.scafacos.available_methods(),
+               'Skipping test: missing ScaFaCoS p3m method')
     def test_actor_exceptions(self):
         system = self.system
 
@@ -71,6 +74,9 @@ class ScafacosInterface(ut.TestCase):
                 prefactor=1, method_name="p3m", method_params={}))
         system.actors.clear()
 
+    @ut.skipIf(not espressomd.has_features('SCAFACOS') or
+               'p3m' not in espressomd.scafacos.available_methods(),
+               'Skipping test: missing ScaFaCoS p3m method')
     def test_actor_coulomb(self):
         system = self.system
 
@@ -90,7 +96,9 @@ class ScafacosInterface(ut.TestCase):
                          {'p3m_cao': '7', 'p3m_r_cut': '1.0',
                           'p3m_grid': '32', 'p3m_alpha': '2.799269'})
 
-    @utx.skipIfMissingFeatures(["SCAFACOS_DIPOLES"])
+    @ut.skipIf(not espressomd.has_features('SCAFACOS_DIPOLES') or
+               'p2nfft' not in espressomd.scafacos.available_methods(),
+               'Skipping test: missing ScaFaCoS p2nfft method')
     def test_actor_dipoles(self):
         system = self.system
 
@@ -191,7 +199,7 @@ class ScafacosInterface(ut.TestCase):
     @utx.skipIfMissingFeatures("LENNARD_JONES")
     @ut.skipIf(not espressomd.has_features('SCAFACOS_DIPOLES') or
                'p2nfft' not in espressomd.scafacos.available_methods(),
-               'Skipping test: missing p2nfft or ewald method')
+               'Skipping test: missing SCAFACOS_DIPOLES or p2nfft method')
     def test_electrostatics_plus_magnetostatics(self):
         # check that two instances of ScaFaCoS can be used
         system = self.system

--- a/testsuite/python/scafacos_interface.py
+++ b/testsuite/python/scafacos_interface.py
@@ -148,12 +148,12 @@ class ScafacosInterface(ut.TestCase):
 
         scafacos_coulomb = espressomd.electrostatics.Scafacos(
             prefactor=0.5,
-            method_name="p3m",
+            method_name="ewald",
             method_params={
-                "p3m_r_cut": 1.0,
-                "p3m_alpha": 2.799269,
-                "p3m_grid": 32,
-                "p3m_cao": 7})
+                "ewald_r_cut": 1.0,
+                "ewald_alpha": 3.861352,
+                "ewald_kmax": 23,
+                "ewald_maxkmax": 100})
         system.actors.add(scafacos_coulomb)
 
         scafacos_dipoles = espressomd.magnetostatics.Scafacos(
@@ -183,7 +183,7 @@ class ScafacosInterface(ut.TestCase):
 
     @utx.skipIfMissingFeatures("LENNARD_JONES")
     @ut.skipIf(not espressomd.has_features('SCAFACOS_DIPOLES') or
-               not {'p2nfft', 'p3m'}.issubset(
+               not {'p2nfft', 'ewald'}.issubset(
                    espressomd.scafacos.available_methods()),
                'Skipping test: missing p2nfft or ewald method')
     def test_electrostatics_plus_magnetostatics(self):
@@ -213,7 +213,7 @@ class ScafacosInterface(ut.TestCase):
 
         self.assertAlmostEqual(fcs_E_coulomb, p3m_E_coulomb, delta=1e-4)
         self.assertAlmostEqual(fcs_E_dipoles, p3m_E_dipoles, delta=1e-4)
-        np.testing.assert_allclose(fcs_forces, p3m_forces, rtol=1e-3)
+        np.testing.assert_allclose(fcs_forces, p3m_forces, rtol=1e-2)
         np.testing.assert_allclose(fcs_torques, p3m_torques, rtol=1e-3)
 
 

--- a/testsuite/python/scafacos_interface.py
+++ b/testsuite/python/scafacos_interface.py
@@ -130,6 +130,7 @@ class ScafacosInterface(ut.TestCase):
             accuracy=1e-5,
             cao=7,
             mesh=48,
+            r_cut=1.88672,
             epsilon="metallic")
         system.actors.add(dp3m)
 
@@ -148,12 +149,18 @@ class ScafacosInterface(ut.TestCase):
 
         scafacos_coulomb = espressomd.electrostatics.Scafacos(
             prefactor=0.5,
-            method_name="ewald",
+            method_name="p2nfft",
             method_params={
-                "ewald_r_cut": 1.0,
-                "ewald_alpha": 3.861352,
-                "ewald_kmax": 23,
-                "ewald_maxkmax": 100})
+                "p2nfft_verbose_tuning": 0,
+                "pnfft_N": "32,32,32",
+                "pnfft_n": "32,32,32",
+                "tolerance_field": "5e-4",
+                "pnfft_window_name": "bspline",
+                "pnfft_m": "4",
+                "p2nfft_ignore_tolerance": "1",
+                "pnfft_diff_ik": "0",
+                "p2nfft_r_cut": "1.0",
+                "p2nfft_alpha": "2.92"})
         system.actors.add(scafacos_coulomb)
 
         scafacos_dipoles = espressomd.magnetostatics.Scafacos(
@@ -183,8 +190,7 @@ class ScafacosInterface(ut.TestCase):
 
     @utx.skipIfMissingFeatures("LENNARD_JONES")
     @ut.skipIf(not espressomd.has_features('SCAFACOS_DIPOLES') or
-               not {'p2nfft', 'ewald'}.issubset(
-                   espressomd.scafacos.available_methods()),
+               'p2nfft' not in espressomd.scafacos.available_methods(),
                'Skipping test: missing p2nfft or ewald method')
     def test_electrostatics_plus_magnetostatics(self):
         # check that two instances of ScaFaCoS can be used

--- a/testsuite/python/scafacos_interface.py
+++ b/testsuite/python/scafacos_interface.py
@@ -181,7 +181,11 @@ class ScafacosInterface(ut.TestCase):
 
         return (ref_E_coulomb, ref_E_dipoles, ref_forces, ref_torques)
 
-    @utx.skipIfMissingFeatures(["SCAFACOS_DIPOLES", "LENNARD_JONES"])
+    @utx.skipIfMissingFeatures("LENNARD_JONES")
+    @ut.skipIf(not espressomd.has_features('SCAFACOS_DIPOLES') or
+               not {'p2nfft', 'p3m'}.issubset(
+                   espressomd.scafacos.available_methods()),
+               'Skipping test: missing p2nfft or ewald method')
     def test_electrostatics_plus_magnetostatics(self):
         # check that two instances of ScaFaCoS can be used
         system = self.system

--- a/testsuite/python/test_checkpoint.py
+++ b/testsuite/python/test_checkpoint.py
@@ -24,6 +24,7 @@ import espressomd
 import espressomd.checkpointing
 import espressomd.electrostatics
 import espressomd.magnetostatics
+import espressomd.scafacos
 import espressomd.virtual_sites
 import espressomd.integrate
 from espressomd.shapes import Sphere, Wall
@@ -367,9 +368,10 @@ class CheckpointTest(ut.TestCase):
             np.testing.assert_almost_equal(p3m_state[key], p3m_reference[key],
                                            err_msg=f'for parameter {key}')
 
-    @utx.skipIfMissingFeatures('SCAFACOS')
-    @ut.skipIf('SCAFACOS' not in modes,
-               "Skipping test due to missing combination.")
+    @ut.skipIf(not espressomd.has_features('SCAFACOS') or
+               'SCAFACOS' not in modes or
+               'p3m' not in espressomd.scafacos.available_methods(),
+               "Skipping test due to missing combination or p3m method.")
     def test_scafacos(self):
         actor = self.get_active_actor_of_type(
             espressomd.electrostatics.Scafacos)
@@ -383,9 +385,10 @@ class CheckpointTest(ut.TestCase):
         for key in reference:
             self.assertEqual(state[key], reference[key], msg=f'for {key}')
 
-    @utx.skipIfMissingFeatures('SCAFACOS_DIPOLES')
-    @ut.skipIf('SCAFACOS' not in modes,
-               "Skipping test due to missing combination.")
+    @ut.skipIf(not espressomd.has_features('SCAFACOS_DIPOLES') or
+               'SCAFACOS' not in modes or
+               'p2nfft' not in espressomd.scafacos.available_methods(),
+               "Skipping test due to missing combination or p2nfft method.")
     def test_scafacos_dipoles(self):
         actor = self.get_active_actor_of_type(
             espressomd.magnetostatics.Scafacos)

--- a/testsuite/scripts/tutorials/test_raspberry_electrophoresis.py
+++ b/testsuite/scripts/tutorials/test_raspberry_electrophoresis.py
@@ -18,29 +18,29 @@
 import unittest as ut
 import importlib_wrapper
 import numpy as np
+np.random.seed(42)
 
 tutorial, skipIfMissingFeatures = importlib_wrapper.configure_and_import(
     "@TUTORIALS_DIR@/raspberry_electrophoresis/raspberry_electrophoresis.py",
-    gpu=True, box_l=20., E=0.25, num_iterations=200, num_steps_per_iteration=250)
+    gpu=True, box_l=20., num_iterations=20, num_steps_per_iteration=20)
 
 
 @skipIfMissingFeatures
 class Tutorial(ut.TestCase):
+    '''Check the raspberry travels a longer distance on the x-axis'''
     system = tutorial.system
 
     def test_trajectory_sample(self):
         trajectory = np.loadtxt('posVsTime_sample.dat')[:, 1:4]
-        # the raspberry should have traveled mostly on the x-axis
-        dist = np.abs(trajectory[-1, :] - trajectory[0, :])
-        self.assertGreater(dist[0], dist[1])
-        self.assertGreater(dist[0], dist[2])
+        x, y, z = np.abs(trajectory[-1, :] - trajectory[0, :])
+        self.assertGreater(x, y)
+        self.assertGreater(x, z)
 
     def test_trajectory_simulated(self):
         trajectory = np.loadtxt('posVsTime.dat')[:, 1:4]
-        # the raspberry should have traveled mostly on the x-axis,
-        # but due to insufficient sampling, it's not always the case
-        dist = np.abs(trajectory[-1, :] - trajectory[0, :])
-        self.assertGreater(dist[0], np.min(dist[1:]))
+        x, y, z = np.abs(trajectory[-1, :] - trajectory[0, :])
+        self.assertGreater(x, y)
+        self.assertGreater(x, z)
 
 
 if __name__ == "__main__":

